resources/aws/subnet.go

```diff
@@ -100,17 +103,23 @@ func (s *Subnet) CreateOrFail() error {
 }
 
 func (s *Subnet) Delete() error {
```

Comment:
Umm... IMO `Subnet` is a data structurem which we fill with the data and send them to AWS API. `Delete` is a method which we run on the single instance.